src/TLuaInterpreter.cpp

```diff
@@ -2971,6 +2971,35 @@ int TLuaInterpreter::setModulePriority(lua_State* L)
     return 0;
 }
 
+int TLuaInterpreter::getModuleSync(lua_State* L)
+{
+  QString moduleName;
+  if (!lua_isstring(L, 1)) {
+    lua_pushstring(L, "getModuleSync: Module must be a string");
+    return lua_error(L);;
+  } else {
+    moduleName = lua_tostring(L, 1);
+  }
+
+  Host& host = getHostFromLua(L);
+  QMap<QString, QStringList> modules = host.mInstalledModules;
+  if (!modules.contains(moduleName)) {
+    lua_pushfstring(L, "getModuleSync: module %s not found", moduleName);
+    return lua_error(L);
+  }
+
+  QStringList moduleSync = host.mModulesToSync;
```

Comment:
Like in the previous, we don't really need a local `moduleSync` either, and since the `QMap<T1, T2>::contains(T1)` method returns a `bool` type directly and we only use the value once we can put things a bit more compactly by changing:
```cpp
    QStringList moduleSync = host.mModulesToSync; 
    if (moduleSync.contains(moduleName)) {
        lua_pusboolean(L, 1);
        return 1;
    } else {
        lua_pushboolean(L, 0);
        return 1;
    }
    return 0;
```
to
```cpp
    lua_pushboolean(L, host.mModulesToSync.contains(moduleName));
    return 1;
```

oh, and `lua_pusboolean(...)` :rofl:

Reply:
:man_facepalming: That was stupid of me, I didn't *read the source, Luke* and hadn't spotted that `Host::mModulesToSync` is a `QStringList` rather than a `QMap` and is **`private`**. So with your new getter `(QStringList) Host::getModulesToSync()` it should, instead, be:
```cpp
    lua_pushboolean(L, host.getModulesToSync().contains(moduleName));
    return 1;
```

Reply:
That being said, this isn't actually true - mModulesToSync is a private member, so we have to either make it public or add a getter function (and now I see below you have accounted for that... You are so helpful!)